Content.Client/_CD/Admin/UI/EventPreferencesPanel.xaml.cs

```diff
@@ -0,0 +1,36 @@
+﻿using Content.Shared.Preferences;
+using Content.Shared.Roles;
+using Robust.Client.AutoGenerated;
+using Robust.Client.UserInterface.CustomControls;
+using Robust.Client.UserInterface.XAML;
+
+namespace Content.Client._CD.Admin.UI;
+
+[GenerateTypedNameReferences]
+public sealed partial class EventPreferencesPanel : DefaultWindow
+{
+    public EventPreferencesPanel()
+    {
+        RobustXamlLoader.Load(this);
+    }
+
+    /// <summary>
+    /// Sets the displayed username of the current player alongside their current selected character's name
+    /// </summary>
+    /// <param name="username">Player's username</param>
+    /// <param name="currentPreferences">Profile preference to fetch the character name from</param>
+    public void SetDetails(string username, HumanoidCharacterProfile currentPreferences)
+    {
+        PlayerLabel.Text = Loc.GetString("cd-eventpreferencepanel-player", ("player", username));
+        CharacterLabel.Text = Loc.GetString(
+            "cd-eventpreferencepanel-character",
+            ("characterName", currentPreferences.Name)
+            );
+    }
+
+    /// <inheritdoc cref="EventPreferences.SetPreferenceList"/>
+    public void SetPreferenceList(HumanoidCharacterProfile userPreferences, IEnumerable<AntagPrototype> listablePreferences)
+    {
+        EventPreferenceControl.SetPreferenceList(userPreferences, listablePreferences);
+    }
```

Comment:
I think these should be condensed into one method because you will always need to call both, and I don't see a situation where they would need to be updated separately.